src/unifiedreport/agent/reportStatic.php

```diff
@@ -284,15 +285,47 @@ function assessmentSummaryTable(Section $section, $otherStatement, $heading)
     return $cell;
   }
 
-
   /**
-   * @brief Get obligation text and re arrange them
-   * @param string $text
-   * @return array $texts
+   * @brief Get obligation text and arrange them to rows and columns
+   * @param string $obligationText
+   * @param string $table
+   * @param string $rowWidth
+   * @param array $firstColLen
+   * @param array $rowTextStyleLeft
+   * @param array $secondColLen
+   * @param array $rowTextStyleRight
+   * @return table columns
    */
-  protected function reArrangeObligationText($text)
+  protected function arrangeObligationsText($obligationText, $table, $rowWidth, $firstColLen, $rowTextStyleLeft, $secondColLen, $rowTextStyleRight)
   {
-    return explode(PHP_EOL, $text);
+    $jsonObligationText = json_decode($obligationText, true);
+    if (json_last_error() === JSON_ERROR_NONE) {
+      foreach ($jsonObligationText as $key => $value) {
+        $table->addRow($rowWidth);
+        $cell = $table->addCell($firstColLen)->addText(htmlspecialchars($key), $rowTextStyleLeft, "pStyle");
+        if (strpos($value, '<br>') !== false) {
+          $cell = $table->addCell($secondColLen);
+          $value = explode('<br>', $value);
+          foreach ($value as $value1) {
+            $cell->addText(htmlspecialchars($value1), $rowTextStyleRight, "pStyle");
```

Comment:
Suggestion to fix the line breaks.

```suggestion
        if (strpos($value, '\n') !== false) {
          $cell = $table->addCell($secondColLen);
          $value = explode('\n', $value);
          foreach ($value as $value1) {
            $value1 = str_replace("\r", "", $value1);
            $cell->addText(htmlspecialchars($value1), $rowTextStyleRight, "pStyle");
```